### DISCUSSION TITLE
Short v-bind on object

### DISCUSSION THREAD

**Discussion by gravitano** [vue-macros/vue-macros#181] [Ideas]
Hi, great work!

I was wondering any plan for creating shorthand for v-bind?

We know that Vue had shorthand for v-bind for single prop:

```
<PostCard :title="title" :description="description" />
```

And we can also use v-bind using object:

```
<PostCard v-bind="{title, description}" />
```

Would be nice if we had shorthand for v-bind on object. Maybe something like or other format syntax is welcome:

```
<PostCard ...{title, description} />
```

Thanks!

**Comment by sxzz** [comment]
Vue Macros welcomes feature proposals. It means if you'd like, you could create a PR for it, and I'd like to accept it. But for me, I don't have too much enthusiasm for this proposal, since I don't think `v-bind` is a directive for frequent daily usage.

**Reply by gravitano** [reply]
Alright.

What about shorthand for single prop binding. I see this feature requested a lot in vue core repo but always rejected. Maybe this could be added via this plugin.

```
<PostCard :title="title" :description="description" />
```

Could be simplify by:

```
<PostCard :title :description />
```

Or

```
<PostCard ::title ::description />
```

Or

```
<PostCard $title $description />
```

Or svelte style:

```
<PostCard {title} {description} />
```

**Reply by sxzz** [reply]
Looks good, but I don't have enough quota for this feature. PR is welcome or I'll try to implement it in the future.

**Comment by zhiyuanzmj** [comment]
I want to create a `simple-bind` feature to do  this:

> For Template
``` vue
<script setup>
const title = ref('')
</script>

<template>
  <div :title />
</template>
```

> For Jsx
``` vue
<script setup lang="tsx">
let title = ref('')

export default <div {title} />
</script> 
```

**Reply by sxzz** [reply]
For template, I think the syntax is acceptable. But in JSX, we already have
```tsx
<div {...{ id }} />
```
The syntax you provide cannot even be parsed by Babel. The scope of Vue Macros is focused on Vue itself.

**Reply by sxzz** [reply]
P.S I think the name would better be `shortBind`

PP.S relate comment https://github.com/vuejs/vue/pull/2877#issuecomment-226618510

**Comment by zhiyuanzmj** [comment]
How about converting `::val` to `v-model:val=val` ？

<img width="660" alt="image" src="https://github.com/vue-macros/vue-macros/assets/32807958/0568e9a5-d742-4a8a-a5de-8950bf0a427f">

**Reply by sxzz** [reply]
It will conflict with `shortVmodel`

**Reply by sxzz** [reply]
I prefer `$` sign, cuz `::=` will be formatted to `:=` when using Prettier.
https://vue-macros.sxzz.moe/macros/short-vmodel.html#known-issues

**Reply by zhiyuanzmj** [reply]
Ok, Maybe I should be compatible with `shortVmodel`.
